src/circuit/listener.js

```diff
@@ -24,7 +34,7 @@ module.exports = (circuit) => {
   listener.listen = async (addr) => {
     const addrString = String(addr).split('/p2p-circuit').find(a => a !== '')
 
-    const relayConn = await circuit._dialer.connectToPeer(multiaddr(addrString))
+    const relayConn = await libp2p.dial(multiaddr(addrString))
     const relayedAddr = relayConn.remoteAddr.encapsulate('/p2p-circuit')
```

Comment:
Something to consider here is that in AutoRelay we are creating a listen addr and then calling `transportManager.listen` to "connect". We will already have a connection per the AutoRelay logic, so that address is going to get thrown out and replaced with whatever this address is. We need to do some reconciliation here, I would find it odd to call listen with one address, and then have my actual address end up being something different.

I think this likely won't be a huge issue for initial relay connections, but if we reconnect to known relays, this address could change. The provided address **should** be the address we end up with, but if it fails for some reason we will dial other known addresses for the peer.

Reply:
> We need to do some reconciliation here, I would find it odd to call listen with one address, and then have my actual address end up being something different.

That is true! I changed this to avoid creating multiple connections to the same peer, but I agree with your point.
So you suggest that we go back and use the `connectToPeer` with a fallback to the dial if we fail?

Reply:
> So you suggest that we go back and use the connectToPeer with a fallback to the dial if we fail?

No. We shouldn't care how we're connected to the peer, really, but we need to be careful about the address we're advertising for this. Something like: Prioritize public addresses, if the connected address matches one of those use it, otherwise pick one of the others.

Again, I think we can do a follow PR for this, so we can focus on clear tests for that.